namespace.go

```diff
@@ -0,0 +1,92 @@
+package neo4j
+
+import "strings"
+
+// Node represents each node in the trie
+type TrieNode struct {
+	children map[rune]*TrieNode
+	mapping  string
+}
+
+// Trie represents a trie and has a pointer to the root node
+type Trie struct {
+	root *TrieNode
+}
+
+// InitTrie will create a new Trie
+func InitTrie() *Trie {
+	result := &Trie{root: &TrieNode{
+		children: make(map[rune]*TrieNode),
+	}}
+	return result
+}
+
+// Insert will take in a word and add it to the trie
+func (t *Trie) Insert(word, mapping string) {
+	currentNode := t.root
+	for _, ch := range word {
+		if _, exists := currentNode.children[ch]; !exists {
+			newChild := &TrieNode{children: make(map[rune]*TrieNode)}
+			currentNode.children[ch] = newChild
+			currentNode = newChild
+		} else {
+			currentNode = currentNode.children[ch]
+		}
+	}
+	currentNode.mapping = mapping
+}
+
+// Search will search if a word is in the trie
+func (t *Trie) Search(word string) (string, string, bool) {
+	currentNode := t.root
+	prefix := strings.Builder{}
+	var lastPrefixMatch string
```

Comment:
You need to keep both lastPrefix and lastPrefixmapping. Return lastPrefix, not prefix. Returning prefix will resulting replacing a prefix without a match.

Reply:
namespaceMappings:
    https://lschema.org/A/: lsa

search(https://lschema.org/A/b) ---> (https://lschema.org/A/, lsa)

Descending down the trie and building the prefix as we go, will guarantee that there will be a matching prefix. Im not sure what the difference is between lastPrefix and prefix, shouldn't there only be one matching prefix?

Reply:
No. Let's say you have

https://lschema.org
https://lschema.org/abc/

Let's say the input is: https://lschema.org/ab. If you return prefix, you'll return https://lschema.org/ab